Soft coordinator shutdown documentation (#14356)
arangodb/arangodb: Documentation/DocuBlocks/Rest/Administration/get_api_shutdown.md

@startDocuBlock get_api_shutdown
@brief initiates the shutdown sequence

@RESTHEADER{GET /_admin/shutdown, Query progress of soft shutdown process, RestShutdownHandler}

@RESTDESCRIPTION

> This call reports progress about a soft coordinator shutdown (see

Reviewer: ```suggestion
<small>Introduced in: v3.7.12, v3.8.1, v3.9.0</small>

This call reports progress about a soft Coordinator shutdown (see
```

documentation of `DELETE /_admin/shutdown?soft=true`). 
In this case, the following types of operations are tracked:

 - AQL cursors (in particular streaming cursors)

>  - Transactions (in particular streaming transactions)

Reviewer: ```suggestion
 - Transactions (in particular stream transactions)
```

>  - Pregel runs (conducted by this coordinator)

Reviewer: ```suggestion
 - Pregel runs (conducted by this Coordinator)
```

 - Ongoing asynchronous requests (using the `x-arango-async: store` HTTP header
 - Finished asynchronous requests, whose result has not yet been
   collected
 - Queued low priority requests (most normal requests)
 - Ongoing low priority requests

This call returns a JSON object of the following shape, indicating the
fact that a soft shutdown is ongoing and the number of active operations
of the various types:

```json
{
  "softShutdownOngoing" : true,
  "AQLcursors" : 0,
  "transactions" : 1,
  "pendingJobs" : 1,
  "doneJobs" : 0,
  "pregelConductors" : 0,
  "lowPrioOngoingRequests" : 1,
  "lowPrioQueuedRequests" : 0,
  "allClear" : false
}
```

Once all numbers have gone to 0, the flag `allClear` is set and the

> coordinator shuts down automatically. This API is not available on
> DBServers and Agents.

Reviewer: ```suggestion
Coordinator shuts down automatically. This API is not available on
DB-Servers and Agents.
```

> This API call was introduced in 3.7.12, 3.8.1 and all versions greater
> or equal to 3.9.

Reviewer: ```suggestion
```

> @RESTRETURNCODES

Reviewer: ```suggestion
@RESTREPLYBODY{softShutdownOngoing,boolean,required,}
Whether a soft shutdown of the Coordinator is in progress.

@RESTREPLYBODY{AQLcursors,number,required,}
Number of AQL cursors that are still active.

@RESTREPLYBODY{transactions,number,required,}
Number of ongoing transactions.

@RESTREPLYBODY{pendingJobs,number,required,}
Number of ongoing asynchronous requests.

@RESTREPLYBODY{doneJobs,number,required,}
Number of finished asynchronous requests, whose result has not yet been collected.

@RESTREPLYBODY{pregelConductors,number,required,}
Number of ongoing Pregel jobs.

@RESTREPLYBODY{lowPrioOngoingRequests,number,required,}
Number of queued low priority requests.

@RESTREPLYBODY{lowPrioQueuedRequests,number,required,}
Number of ongoing low priority requests.

@RESTREPLYBODY{allClear,boolean,required,}
Whether all active operations finished.

@RESTRETURNCODES
```
